lua/apisix/balancer.lua

```diff
@@ -122,6 +123,38 @@ local function fetch_healthchecker(upstream, healthcheck_parent, version)
 end
 
 
+local function create_chash_hash_key(ctx, upstream)
+    local key = upstream.key
+    local hash_on = upstream.hash_on
+    local chash_key
+
+    -- hash_on is nil, compatible with old version, default vars
+    if not hash_on then
+        hash_on = "vars"
+    end
+
+    if hash_on == "consumer" then
+        chash_key = ctx.consumer_id
+    elseif hash_on == "vars" then
+        chash_key = ctx.var[key]
+    elseif hash_on == "header" then
+        chash_key = ngx.req.get_headers()[key]
```

Comment:
`chash_key = ctx.var["http_" .. key]` is better.

http://nginx.org/en/docs/http/ngx_http_core_module.html#var_http_

Reply:
get